modulemd/v2/modulemd-yaml-util.c

```diff
@@ -382,6 +386,22 @@ modulemd_yaml_parse_string (yaml_parser_t *parser, GError **error)
   return g_strdup ((const gchar *)event.data.scalar.value);
 }
 
+
+guint64
+modulemd_yaml_parse_uint64 (yaml_parser_t *parser, GError **error)
+{
+  MMD_INIT_YAML_EVENT (event);
+
+  YAML_PARSER_PARSE_WITH_EXIT_INT (parser, &event, error);
+  if (event.type != YAML_SCALAR_EVENT)
+    {
+      MMD_YAML_ERROR_EVENT_EXIT_INT (error, event, "String was not a scalar");
+    }
+
+  return g_ascii_strtoull ((const gchar *)event.data.scalar.value, NULL, 10);
```

Comment:
This would not be according to the documentation for this function, I believe.
Specifically, with the error cases: this function might return `G_MAXUINT64` or zero in case of errors.
I am wondering whether we should be interpreting those values and the corresponding `errno` values and then set an error?
Or, alternatively, document this?

Reply:
Agreed